state/state-tree/src/state_tree.rs

```diff
@@ -350,8 +378,10 @@ where
         if node_key == &*SPARSE_MERKLE_PLACEHOLDER_HASH {
             return Ok(Some(Node::new_null()));
         }
-        if let Some(n) = self.cache.change_set.node_batch.get(node_key).cloned() {
-            return Ok(Some(n));
+        for change_set in self.cache.change_set_list.iter().rev() {
```

Comment:
> 感觉这里不对，应该从最后的 changeset 向前遍历，并且还要考虑删除的情况。

这里是因为remove的时候,往里面插入了None，
```
   /// Remove key_hash's data.
    /// this will not compute new root hash,
    /// Use `commit` to recompute the root hash.
    pub fn remove(&self, key: &K) {
        self.updates.write().insert(key.clone(), None);
    }
```

Reply:
哦，对的。那 get 的时候从后面遍历就是对的。